src/poetry/utils/env/env_manager.py

```diff
@@ -405,10 +405,34 @@ def remove(self, python: str) -> Env:
                     del envs[base_env_name]
                     envs_file.write(envs)
 
-        self.remove_venv(venv_path)
+        self.remove_venv_dir(venv_path)
 
         return VirtualEnv(venv_path, venv_path)
 
+    def remove_venv(self, venv: VirtualEnv) -> Env:
+        venv_path = self._poetry.config.virtualenvs_path
+        cwd = self._poetry.file.path.parent
+        envs_file = TOMLFile(venv_path / self.ENVS_FILE)
```

Comment:
```suggestion
        envs_file = TOMLFile(venv.path.parent / self.ENVS_FILE)
```

Shouldn't that be enough to get the location?